### PR TITLE
Add ssh analyzer 

### PR DESCRIPTION
**Pull request by osogi**
Add BackGroundBuilder and analyzer for it:
- now it is able to build binaries and launch it for testing simultaneously
- latter planned to add a background patcher to them

Add ssh analyzer (fix: #53):
- can connect by password or key
- send binaries (tests), launch them and collect data
- need root user at test machine for correct work

### CODE REVIEW

File: src/helpers/backGroundBuilder.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from dataclasses import dataclass
+import logging
+import os
+from argparse import Namespace
+from pathlib import Path
+from queue import Queue
+import threading
```
Comment (RoketFlame) [inline src/helpers/backGroundBuilder.py:7]: Maybe asicio will be better?

File: src/analyzers/collectors/sshCollector.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+class SshCollector:
+    def __init__(self, settings: Namespace, bin_dir: Path | None = None):
+        self.settings = settings
+        self.max_test_launches = settings.__dict__.get("max_test_launches", -1)
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(self.settings.log_level)
+        self.is_tmp_created = False
+
+        host = settings.__dict__.get("host", "127.0.0.1")
```
Comment (RoketFlame) [inline src/analyzers/collectors/sshCollector.py]: Sould be rewrited with vars

File: src/analyzers/collectors/perfParser.py [hunk 3]
```diff
@@ -65,11 +65,9 @@ def output_to_dict(output: str) -> Dict[str, str]:
         return data_dict
 
     @staticmethod
-    def test_res_to_data(out_tup: TestRes):
-        stream, is_full = out_tup
-        dic: Dict[str, str] = {}
-        if stream is not None:
-            dic = PerfParser.output_to_dict(stream.decode())
+    def test_res_to_data(res: TestRes):
+        stream, is_full = res
```
Comment (RoketFlame) [inline src/analyzers/collectors/perfParser.py:69]: Rename stream to bytes

**Review by RoketFlame — CHANGES_REQUESTED** [review]

File: src/analyzers/collectors/sshCollector.py (hunk 2 above)
Comment (osogi) [reply src/analyzers/collectors/sshCollector.py]: Maybe create separate PR with commit which fix all `__dict__` -> `vars()`

File: src/helpers/backGroundBuilder.py (hunk 1 above)
Comment (RoketFlame) [reply src/helpers/backGroundBuilder.py:7]: No, it is your responsobilty

File: src/analyzers/collectors/sshCollector.py (hunk 2 above)
Comment (RoketFlame) [reply src/analyzers/collectors/sshCollector.py]: Good idea

**Review by vkutuev — APPROVED** [review]

File: src/analyzers/collectors/perfCollector.py [hunk 4]
```diff
@@ -37,6 +37,8 @@ def execute_test(self, execute_line: List[str], timeout: float) -> TestRes:
                 pass
             if proc.returncode is not int:  # some bug: if send signal to proc, it ret code will be one
```
Comment (vkutuev) [inline src/analyzers/collectors/perfCollector.py:38]: Its

**Review by RoketFlame — APPROVED** [review]

**Pull request merged**